### PR TITLE
[improve](scan) support lower the thread priority of scan thread

### PR DESCRIPTION
**Pull request by eldenmoon**
The configuration item is used to lower the priority of the scanner thread, typically employed to ensure CPU scheduling for write operations.

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

File: be/src/util/thread.cpp [hunk 1]
```diff
@@ -93,6 +95,8 @@ class ThreadMgr {
 
 #ifndef __APPLE__
     static void set_idle_sched(int64_t tid);
+
+    static void set_low_priority(int64_t tid);
```
Comment (xiaokang) [inline be/src/util/thread.cpp:99]: can we reuse set_idle_sched?

File: be/src/common/config.cpp [hunk 2]
```diff
@@ -1088,6 +1088,8 @@ DEFINE_Int16(bitmap_serialize_version, "1");
 // the count of thread to group commit insert
 DEFINE_Int32(group_commit_insert_threads, "10");
 
+DEFINE_mBool(enable_scan_thread_low_thread_priority, "false");
```
Comment (xiaokang) [inline be/src/common/config.cpp:1091]: DEFINE_Int32(scan_thread_nice_value, "0")

File: be/src/util/thread.cpp [hunk 3]
```diff
@@ -174,6 +178,27 @@ void ThreadMgr::set_idle_sched(int64_t tid) {
...
+    // In the current implementation, each unit of difference in the nice values of two
+    // processes results in a factor of 1.25 in the degree to which the
+    // scheduler favors the higher priority process.  This causes very
+    // low nice values (+19) to truly provide little CPU to a process
+    // whenever there is any other higher priority load on the system,
+    // and makes high nice values (-20) deliver most of the CPU to
+    // applications that require it (e.g., some audio applications).
+
+    // Choose 5 as lower priority value, default is 0
+    constexpr static int s_low_priority_nice_value = 5;
```
Comment (xiaokang) [inline be/src/util/thread.cpp:196]: use scan_thread_nice_value instead of fixed value

File: be/src/util/thread.cpp (hunk 1 above)
Comment (eldenmoon) [reply be/src/util/thread.cpp:99]: no they are different sys api

File: be/src/util/thread.cpp (hunk 3 above)
Comment (eldenmoon) [reply be/src/util/thread.cpp:196]: ok

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**